### DISCUSSION TITLE
Add weasyprint as an output method

### DISCUSSION THREAD

**Discussion by sbts** [ledgersmb/LedgerSMB#5843] [Design]
Now that debian jessie has wkhtmltopdf 0.12.1 the fairly important header and footer capability is now available.

This means it's time to add an output processor ;-)

**Comment by freelock** [comment]
wkhtmltopdf is pretty much abandoned at this point, does not support modern CSS3 features like CSS Grid/CSS Flexbox. Meaning that theming a template involves using non-standard CSS from 9 years ago, if you want anything more sophisticated than a table layout.

After surveying PDF generation options, two stand out as viable in early 2020: 

1. Headless Chrome (possibly via Puppeteer or similar wrapper API)
2. Weasyprint (a Python library)

In my evaluation, Weasyprint seems to be better -- lighter weight, a lot more PDF-specific options such as table of contents, odd/even footers, etc. and recommend we work towards implementing that instead of wkhtmltopdf.

**Comment by ehuelsmann** [comment]
Although not implemented as proposed, this should now be easily implemented as a plugin: `LedgerSMB::Template::Plugin::Weasy` . 

Closing this item with the comment "patches welcome".
